src/raster/buffer.rs

```diff
@@ -0,0 +1,86 @@
+use crate::raster::GdalType;
+
+#[cfg(feature = "ndarray")]
+use ndarray::Array2;
+
+/// A 2-D array backed by it's `size` (cols, rows) and a row-major `Vec<T>` and it's dimensions.
+#[derive(Debug, Clone, PartialEq)]
+pub struct Buffer<T> {
```

Comment:
I wonder if we should call it `Buffer2`.

Reply:
I mean, just in case we want to do something fancy for the MD API (which I didn't look at too closely).

Reply:
Well, I don't know if it's really a good idea, just thought it might make sense.